### ISSUE TITLE
Interlock between the waypoints

### ISSUE THREAD

**Issue by hiradg** [UoBFlightLab/mav_missions#30]
An interlock feature that prevents vehicles to fly to the same waypoint at the same time

**Comment by arthurrichards77** [comment]
I have an issue when flying the "out and back" example in SITL.  I send a drone round the mission and land it using the "TOPAD" clearance at the end.  However, I can't seem to get "TOPAD" back once the drone lands.  It lands, disarms, and resets its counter to WP1, but it doesn't clear the TOPAD button for other drones.  I have tried clicking on `XXXXX` to withdraw the clearance, but it still keeps `TOPAD` locked out for others.

**Comment by arthurrichards77** [comment]
> I have an issue when flying the "out and back" example in SITL. I send a drone round the mission and land it using the "TOPAD" clearance at the end. However, I can't seem to get "TOPAD" back once the drone lands. It lands, disarms, and resets its counter to WP1, but it doesn't clear the TOPAD button for other drones. I have tried clicking on `XXXXX` to withdraw the clearance, but it still keeps `TOPAD` locked out for others.

It does clear if I take off again with the drone that was blocking `TOPAD`.

**Comment by hiradg** [comment]
By design the clearance tab is disable for the following two reasons:

1. The drone is landed at the pad and that pad is occupied unless the drone is cleared to take off
2. The button interlock looks at whether the drone has accepted a clearance to move since the last clearance. If not it we assume its at the previous target waypoint and will keep that clearance button disabled.

 I do see benefit of having a reset tab for situations where the drone is manually removed from the pad by the operator.

**Comment by arthurrichards77** [comment]
Understood.  Please consider moving the interlock functionality to the "JumpController" class rather than living in the GUI.  I think that will make it easier to maintain and test.  JumpController should offer a `request_clearance(drone_id, clearance)` method that returns `current_clearance, status`.  The GUI just calls the request method and then colours buttons according to the response.  The logic, which will need to grow as we introduce more complicated patterns.  Ought to live elsewhere.

**Comment by arthurrichards77** [comment]
Superseded by block concept
